packages/cli/add/commands/project.js

```diff
@@ -0,0 +1,87 @@
+import prompts from 'prompts';
+import path from 'path';
+import fs from 'fs';
+import ts from 'typescript';
+
+import {projectTemplate} from '../templates.js';
+import {createDir} from '../../utils/createDir.js';
+import {createFile} from '../../utils/createFile.js';
+
+export async function addProject() {
+  const response = await prompts({
+    type: 'text',
+    name: 'projectName',
+    message: 'Project Name',
+    initial: 'myProject',
+    validate: value =>
+      isValidProjectName(value) ? 'Project already exists.' : true,
+  });
+  await createProject(response.projectName);
+}
+
+/* Is it OK to check if the project exists using this? */
+function isValidProjectName(projectName) {
+  return fs.existsSync(path.resolve('src', `${projectName}.ts`));
+}
+
+function isProjectArray(node) {
+  if (!ts.isArrayLiteralExpression(node)) return false;
+  const projectPropertyAssignment = node.parent;
+  if (
+    !(
+      projectPropertyAssignment &&
+      ts.isPropertyAssignment(projectPropertyAssignment) &&
+      ts.isIdentifier(projectPropertyAssignment.name) &&
+      projectPropertyAssignment.name.escapedText === 'project'
+    )
+  )
+    return false;
+  return true;
```

Comment:
This could be simplified to:
```suggestion
  return (
    ts.isArrayLiteralExpression(node) &&
    node.parent &&
    ts.isPropertyAssignment(node.parent) &&
    ts.isIdentifier(node.parent.name) &&
    node.parent.name.escapedText === 'project'
  );
```